src/proj_equirect.jl

```diff
@@ -0,0 +1,555 @@
+
+    ProjEquiRect{T}(Ny, Nx, θspan, φspan, θ, φ, θ∂, φ∂, Ω, storage)
+
+end
+
+function ProjEquiRect(; T=Float32, storage=Array, kwargs...)
+
+    arg_error() = error("Constructor takes either (θ, φ, θ∂, φ∂) or (Ny, Nx, θspan, φspan) keyword arguments.")
+    
+    if all(haskey.(Ref(kwargs), (:θ, :φ, :θ∂, :φ∂)))
+        !any(haskey.(Ref(kwargs), (:Ny, :Nx, :θspan, :φspan))) || arg_error()
+        @unpack (θ, φ, θ∂, φ∂) = kwargs
+    elseif all(haskey.(Ref(kwargs), (:Ny, :Nx, :θspan, :φspan)))
+        !all(haskey.(Ref(kwargs), (:θ, :φ, :θ∂, :φ∂))) || arg_error()
+        @unpack (Ny, Nx, θspan, φspan) = kwargs
+        θ, θ∂ = @ondemand(CirculantCov.θ_grid)(; θspan, N=Ny, type=:equiθ)
+        φ, φ∂ = @ondemand(CirculantCov.φ_grid)(; φspan, N=Nx)
+    else
+        arg_error()
+    end
+
+    ProjEquiRect(θ, φ, θ∂, φ∂, real_type(T), storage)
+
+end
+
+
+
+# Field Basis
+# ================================================
+"""
+Jperm(ℓ::Int, n::Int) return the column number in the J matrix U^2
+where U is unitary FFT. The J matrix looks like this:
+
+|1   0|
+|  / 1|
+| / / |
+|0 1  |
+
+"""
+function Jperm end
+
+function Jperm(ℓ::Int, n::Int)
+    @assert 1 <= ℓ <= n
+    ℓ==1 ? 1 : n - ℓ + 2
+end
+
+# AzFourier <-> Map
+function AzFourier(f::EquiRectMap)
+    nφ = f.Nx
+    EquiRectAzFourier(m_rfft(f.arr, 2) ./ √nφ, f.proj)
+end
+
+function Map(f::EquiRectAzFourier)
+    nφ = f.Nx
+    EquiRectMap(m_irfft(f.arr, nφ, 2) .* √nφ, f.proj)
+end
+
+# QUAzFourier <-> QUMap
+function QUAzFourier(f::EquiRectQUMap)
+    nθ, nφ = f.Ny, f.Nx
+    qiumap = complex.(f.Qx, f.Ux) 
+    Uf = m_fft(qiumap, 2) ./ √nφ
+    f▫ = similar(Uf, 2nθ, nφ÷2+1)
+    for ℓ = 1:nφ÷2+1
+        if (ℓ==1) | ((ℓ==nφ÷2+1) & iseven(nφ))
+            f▫[1:nθ, ℓ]     .= Uf[:,ℓ]
+            f▫[nθ+1:2nθ, ℓ] .= conj.(Uf[:,ℓ])
+        else
+            f▫[1:nθ, ℓ]     .= Uf[:,ℓ]
+            f▫[nθ+1:2nθ, ℓ] .= conj.(Uf[:,Jperm(ℓ,nφ)])
+        end
+    end
+    EquiRectQUAzFourier(f▫, f.proj)
+end
+
+function QUMap(f::EquiRectQUAzFourier)
+    nθₓ2, nφ½₊1 = size(f.arr)
+    nθ, nφ = f.Ny, f.Nx
+    @assert nφ½₊1 == nφ÷2+1
+    @assert 2nθ   == nθₓ2
+
+    pθk = similar(f.arr, nθ, nφ)
+    for ℓ = 1:nφ½₊1
+        if (ℓ==1) | ((ℓ==nφ½₊1) & iseven(nφ))
+            pθk[:,ℓ] .= f.arr[1:nθ,ℓ]
+        else
+            pθk[:,ℓ]  .= f.arr[1:nθ,ℓ]
+            pθk[:,Jperm(ℓ,nφ)] .= conj.(f.arr[nθ+1:2nθ,ℓ])
+        end
+    end
+    qiumap = m_ifft(pθk, 2) .* √nφ
+    EquiRectQUMap(cat(real(qiumap), imag(qiumap), dims=3), f.proj)
+end
+
+# QUESTION: Any interest in keeping these? 
+# I use them a lot while I'm playing around and testing but ..
+# f[:] conflicts with the AbstractArray characterization and 
+# 
+Base.getindex(f::EquiRectS0, ::typeof(!)) = AzFourier(f).arr
+Base.getindex(f::EquiRectS2, ::typeof(!)) = QUAzFourier(f).arr
+Base.getindex(f::EquiRectS0, ::Colon)     = Map(f).arr
+Base.getindex(f::EquiRectS2, ::Colon)     = (qu=QUMap(f); complex.(qu.Qx,qu.Ux))
```

Comment:
Unfortunatley I think these have to go, but there's basically already a "standard" way to do this, see: https://cosmicmar.com/CMBLensing.jl/stable/05_field_basics/#Properties-and-indices-1 

So basically your `f[:]` becomes `f[:Ix]`, `f[:Qx]`, etc... (this actually already works for EquiRectFields since its using the generic definition [here](https://github.com/marius311/CMBLensing.jl/blob/ceb58d5d826e07fcd9b2231b46f67f9ac474c3c9/src/proj_cartesian.jl#L78-L87)) and your `f[!]` becomes something like `f[:Il]`, or `f[:Ql]` (except I think we should keep the relation `:Il <-> Fourier` so since here its `AzFourier`, I would do a different symbol, maybe `:Im` to reflect that the result is a function of `m` (and `θ`) rather than an `l`-vector ? 

I can do this this.

Reply:
Done. I went with `:Px` and `:Pl` in the end.